pallets/block-rewards/src/lib.rs

```diff
@@ -0,0 +1,449 @@
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		NewEpoch {
+			collator_reward: T::Balance,
+			beneficiary_reward: T::Balance,
+			last_changes: EpochChanges<T>,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Limit of max calls with same id to [`Pallet::set_group_weight()`] or
+		/// [`Pallet::set_currency_group()`] reached.
+		MaxChangesPerEpochReached,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Claims the reward the associated to a currency.
+		/// The reward will be transferred to the target account.
+		#[pallet::weight(T::WeightInfo::claim_reward())]
+		#[transactional]
+		pub fn claim_reward(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			account_id: T::AccountId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			T::Rewards::claim_reward((T::Domain::get(), currency_id), &account_id).map(|_| ())
+		}
+
+		/// Admin method to set the reward amount for a collator used for the next epochs.
+		/// Current epoch is not affected by this call.
+		#[pallet::weight(T::WeightInfo::set_distributed_reward())]
+		pub fn set_collator_reward(
+			origin: OriginFor<T>,
+			collator_reward_per_session: T::Balance,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			NextEpochChanges::<T>::mutate(|changes| {
+				changes.collator_reward = Some(collator_reward_per_session);
+			});
+
+			Ok(())
+		}
+
+		/// Admin method to set the inflation amount for the Beneficiary used for the next epochs.
+		/// Current epoch is not affected by this call.
+		#[pallet::weight(T::WeightInfo::set_distributed_reward())]
+		pub fn set_beneficiary_reward(
+			origin: OriginFor<T>,
+			beneficiary_reward_per_session: T::Balance,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			NextEpochChanges::<T>::mutate(|changes| {
+				changes.beneficiary_reward = Some(beneficiary_reward_per_session)
+			});
+
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	/// Mint default amount of stake for target address and deposit stake.
+	/// Enables receiving rewards onwards.
+	fn do_init_collator(who: &T::AccountId) -> DispatchResult {
+		T::Currency::mint_into(
+			T::CollatorCurrencyId::get(),
+			who,
+			T::DefaultCollatorStake::get(),
+		)?;
+		T::Rewards::deposit_stake(
+			(T::Domain::get(), T::CollatorCurrencyId::get()),
+			who,
+			T::DefaultCollatorStake::get(),
+		)
+	}
+
+	/// Withdraw currently staked amount for target address and immediately burn it.
+	/// Disables receiving rewards onwards.
+	fn do_exit_collator(who: &T::AccountId) -> DispatchResult {
+		let amount =
+			T::Rewards::account_stake((T::Domain::get(), T::CollatorCurrencyId::get()), who);
+		T::Rewards::withdraw_stake(
+			(T::Domain::get(), T::CollatorCurrencyId::get()),
+			who,
+			amount,
+		)?;
+		T::Currency::burn_from(T::CollatorCurrencyId::get(), who, amount).map(|_| ())
```

Comment:
We are burning because `withdraw_stake` frees the currency from its holding state such that it is transferrable by default. Additionally, there is no reason for someone holding the artificial staking currency without being a collator.